worldedit-sponge/src/main/java/com/sk89q/worldedit/sponge/SpongeWorld.java

```diff
@@ -142,6 +152,43 @@ public boolean setBlock(Vector position, BaseBlock block, boolean notifyAndLight
 
     @Override
     public boolean regenerate(Region region, EditSession editSession) {
+        try {
+            Server server = Sponge.getServer();
+            WorldProperties tempWorldProperties;
+            try {
+                tempWorldProperties = server.createWorldProperties("worldedittemp", this.regenerationArchetype);
+            } catch (IOException e) {
+                e.printStackTrace();
+                return false;
+            }
+
+            Optional<World> tempWorldOpt = server.loadWorld(tempWorldProperties);
+            if (!tempWorldOpt.isPresent()) {
+                return false;
```

Comment:
Might want to include a log line here, just so it's easy to know why regen isn't working.